addons/web/controllers/json.py

```diff
@@ -244,6 +291,61 @@ def get_action_triples_():
         context.update(safe_eval(action.context, eval_context))
         return action, context, eval_context, record_id
 
+    def _web_json_2_rpc_readonly(self):
+        try:
+            model_name = request.httprequest.args['model']
+            method_name = request.httprequest.args['method']
+            Model = request.env[model_name]
+        except KeyError:
+            return False
+        for cls in Model.mro():
+            method = getattr(cls, method_name, None)
+            if method is not None and hasattr(method, '_readonly'):
+                return method._readonly
+        return False
+
+    @http.route(
+        '/json/2/<model>/<method>',
+        #methods=['POST'],  # must accept all methods, otherwise /json/<path:subpath> takes over
+        auth='bearer',
+        type='/json/2/rpc',
+        readonly=_web_json_2_rpc_readonly,
+        save_session=False
+    )
+    def web_json_2_rpc(
+        self,
+        model: str,
+        method: str,
+        ids: Sequence[int] = (),
+        kwargs: Mapping[str, Any] = frozendict(),
+        context: Mapping[str, Any] = frozendict(),
+    ):
+        if request.httprequest.method != 'POST':
```

Comment:
When readonly, we could accept GET as well.

Reply:
bad idea